extensions/wikia/Flags/controllers/FlagsController.class.php

```diff
@@ -7,10 +7,42 @@
 class FlagsController extends WikiaApiController {
 
 	private
+		$model,
 		$params,
 		$status = false;
 
 	/**
+	 * Generates html contents for Flags modal for editing flags
+	 */
+	public function modal() {
+		global $wgUser;
+
+		$pageId = $this->request->getVal( 'pageId' );
+		if ( empty( $pageId ) ) {
+			return null;
+		}
+
+		$flagsWikiaResponse = $this->app->sendRequest( 'FlagsController', 'getFlagsForPageForEdit', [
+			'pageId' => $pageId
+		] );
+
+		$flags = $flagsWikiaResponse->getData();
+
+		$html = \HandlebarsService::getInstance()->render(
```

Comment:
We compare PHP/Mustache/Handlebars template engines performance (https://docs.google.com/document/d/1_s24sj1yWRVzlng9Q2b7lYhWO8hwfpzLAEvb0df5k0Y/edit#heading=h.kvbvi3mr8dx1) and we resigned from Handlebars becasuse is slowest one (almost 200 slower than PHP). So maybe you can use PHP templates or change structure for Mustache.